### PR TITLE
Added getting quotas from other chips, Fixes: #825

### PR DESCRIPTION
**Pull request by adamnejm**

### CODE REVIEW

**Comment by DaDamRival** [comment]
I think having them as entity methods would be better and maby also add getName and getAuthor.

**Comment by adamnejm** [comment]
good idea, go do

**Comment by Damianu** [comment]
Neat, only thing I'd add is make getName() support e2 too

File: lua/starfall/libs_sh/entities.lua [hunk 1]
```diff
@@ -1026,4 +1027,77 @@ function ents_methods:getCreationTime()
...
+-- @shared
+-- @return Average CPU Time of the buffer of the specified starfall.
+function ents_methods:getQuotaAverage()
+	local ent = getent(self)
+	if not ent.Starfall then SF.Throw("The entity isn't a starfall chip", 2) end
+	
+	return ent.instance:movingCPUAverage()
+end
+
+--- Gets the CPU Time max of the specified starfall of the specified starfall.
```
Comment (Damianu) [inline lua/starfall/libs_sh/entities.lua]: Typo

**Comment by adamnejm** [comment]
In case ExpAdv3 has the same structure, be sure to exclude it

**Comment by thegrb93** [comment]
Anytime you grab data off an arbitrary entity, you have to be sure the data is sanitary. The likelihood of a non-starfall having these fields is small, but still could happen. I can add those tomorrow.

**Comment by Damianu** [comment]
> 
> 
> ~Anytime you grab data off an arbitrary entity, you have to be sure the data is sanitary. The likelihood of a non-starfall having these fields is small, but still could happen. I can add those tomorrow.~
> 
> ent.Starfall I guess covers it.

Was about to say if entity has Starfall property on it, it's kinda its own fault :D

**Comment by thegrb93** [comment]
I think 'Starfall' should be IsStarfall or something less conflicty

**Comment by Damianu** [comment]
> 
> 
> I think 'Starfall' should be IsStarfall or something less conflicty

Might check class, however I think both are overkill, it's unlikely something will have prop named starfall on it, and if it does it can be fixed later.

**Comment by thegrb93** [comment]
Most likely. Looks fine but I'll review it when I get some sleep

**Comment by thegrb93** [comment]
There's already entities.getChipName, the others should be entities.getChipAuthor, getChip* and should be compatible with E2.

**Comment by thegrb93** [comment]
Looks like E2 doesn't have an author getter.

**Comment by thegrb93** [comment]
Other starfall's cpu_total will not update in the context of the current starfall so removed getQuotaUsed.

**Comment by thegrb93** [comment]
Test and see how this works

**Pull request merged**